src/Microsoft.Health.Fhir.Api/Features/BackgroundJobService/JobFactory.cs

```diff
@@ -18,22 +20,33 @@ public class JobFactory : IJobFactory
     {
         private readonly Dictionary<int, Func<IJob>> _jobFactoryLookup;
 
-        public JobFactory(IEnumerable<Func<IJob>> jobFactories)
+        public JobFactory(IEnumerable<Func<IJob>> jobFactories, ILogger<JobFactory> logger)
         {
             EnsureArg.IsNotNull(jobFactories, nameof(jobFactories));
 
             _jobFactoryLookup = new Dictionary<int, Func<IJob>>();
 
             foreach (Func<IJob> jobFunc in jobFactories)
             {
-                var instance = jobFunc.Invoke();
-                if (instance.GetType().GetCustomAttribute(typeof(JobTypeIdAttribute), false) is JobTypeIdAttribute jobTypeAttr)
+                try
```

Comment:
Why wrap this in a try/catch here? This looks like it would allow the JobFactory to be created successfully without all of the job types successfully added. This should be caught during testing as the affected job type wouldn't run, but I don't see why we would want this to succeed if there are broken job types.

Reply:
What I found was, the Queues determine the job types that are needed, Job Types can be registered that aren't used. In this examples Import was in core and registered, but this isn't ever used by a Cosmos instance